lib/Doctrine/Common/Lexer/Token.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Common\Lexer;
+
+use ArrayAccess;
+use ReturnTypeWillChange;
+
+use function in_array;
+
+/**
+ * @template T of string|int
+ * @implements ArrayAccess<string,mixed>
+ */
+final class Token implements ArrayAccess
```

Comment:
Let's also implement `IteratorAggregate` and `Countable` for array compatibility? We should also add tests that cover the BC layer.

Reply:
`IteratorAggregate` is not needed to get the expected result in a foreach, I will demonstrate that with a test.

Reply:
Here is the test:

```php
    public function testIterationStillWorks(): void
    {
        $token = new Token('foo', 'string', 1);

        $tokenAsArray = [];

        foreach ($token as $key => $value) {
            $tokenAsArray[$key] = $value;
        }

        self::assertSame(['value' => 'foo', 'type' => 'string', 'position' => 1], $tokenAsArray);
    }
```

It works, but I'm not going to commit it because static analysis is unhappy about it, and because there should be no deprecation for that scenario, since, it's going to continue working after the BC layer is removed.